aspnet/fundamentals/static-files.rst

```diff
@@ -47,132 +54,95 @@ Now, let's say that you have a project hierarchy where the static files you wish
 
     - test.png
 
-In order for the user to access test.png, you can configure the static files middleware as follows:
+For a request to access *test.png*, configure the static files middleware as follows:
 
-.. code-block:: c#
-  :emphasize-lines: 5-9
+.. literalinclude:: static-files/sample/Startup.cs
+  :language: c#
+  :lines: 38-39,53-62
+  :emphasize-lines: 7-12
+  :dedent: 8
+
+A request to ``http://<app>/StaticFiles/test.png`` will serve the *test.png* file.
 
-  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
-  {
-    ...
-    // Add MyStaticFiles static files to the request pipeline.
-    app.UseStaticFiles(new StaticFileOptions()
-    {
-        FileProvider = new PhysicalFileProvider(@"D:\Source\WebApplication1\src\WebApplication1\MyStaticFiles"),
-        RequestPath = new PathString("/StaticFiles")
-    });
-    ...
+Static file authorization
+-------------------------
 
-At this point, if the user enters an address of ``http://<yourApp>/StaticFiles/test.png``, the ``test.png`` image will be served.
+The static file module provides **no** authorization checks. Any files served by it, including those under *wwwroot* are publicly available. To serve files based on authorization:
+
+- Store them outside of *wwwroot* and any directory accessible to the static file middleware **and**
+- Serve them through a controller action, returning a :dn:class:`~Microsoft.AspNetCore.Mvc.FileResult` where authorization is applied
 
 Enabling directory browsing
 ---------------------------
 
-Directory browsing allows the user of your Web app to see a list of directories and files within a specified directory (including the root). By default, this functionality is not available such that if the user attempts to display a directory within an ASP.NET Web app, the browser displays an error. To enable directory browsing for your Web app, call the ``UseDirectoryBrowser`` extension method from  ``Startup.Configure`` as follows:
-
-.. code-block:: c#
-  :emphasize-lines: 5
+Directory browsing allows the user of your web app to see a list of directories and files within a specified directory. Directory browsing is disabled by default for security reasons (see Considerations_). To enable directory browsing, call the :dn:method:`~Microsoft.AspNetCore.Builder.DirectoryBrowserExtensions.UseDirectoryBrowser` extension method from  ``Startup.Configure``:
 
-  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
-  {
-    ...
-    // Turn on directory browsing for the current directory.
-    app.UseDirectoryBrowser();
-    ...
+.. literalinclude:: static-files/sample/StartupBrowse.cs
+  :language: c#
+  :lines: 38-39,52-68
+  :dedent: 8
 
-The following figure illustrates the results of browsing to the Web app's ``images`` folder with directory browsing turned on:
+The code above allows directory browsing of the *wwwroot/images* folder with links to each file:
```

Comment:
This is a bit ambiguous because subdirectories of wwwroot are already browsable. Pick a different folder, or say that you did this to make them browsable via the `/MyImages` url.